emcc.py

```diff
@@ -2972,13 +2974,15 @@ def check_bad_eq(arg):
       settings_changes.append('PTHREADS_PROFILING=1')
       newargs[i] = ''
     elif newargs[i] == '-fno-exceptions':
-      settings_changes.append('DISABLE_EXCEPTION_CATCHING=1')
-      settings_changes.append('DISABLE_EXCEPTION_THROWING=1')
+      shared.Settings.DISABLE_EXCEPTION_CATCHING = 1
+      shared.Settings.DISABLE_EXCEPTION_THROWING = 1
+      shared.Settings.EXCEPTION_HANDLING = 0
```

Comment:
isn't it better to use the `settings_changes` mechanism, so that `-fno-exceptions` mixes properly with `-s` of the relevant flags?

Reply:
Oh, sorry, I remembered that we made `EXCEPTION_HANDLING` an internal setting, so we shouldn't use `settings_changes` for it. So the handling of `-fno-exceptions` on these lines looks good.

Reply:
Yes I had to use these because we moved `EXCEPTION_HANDLING` to settings_internal.js.